x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -1552,8 +1552,11 @@ export class Embeddable
         )
       )
     ).forEach((dataView) => indexPatterns.push(dataView));
+    const prevInternalDataViews = this.internalDataViews;
 
-    this.internalDataViews = uniqBy(indexPatterns, 'id');
+    // making sure to not run into the race condition that the data source has the previous data view id
+    // causing to throw an exception that's not necessary
+    this.internalDataViews = uniqBy([...indexPatterns, ...prevInternalDataViews], 'id');
```

Comment:
I don't know if we can commit it this way, but it appears to resolve the race condition in the manual testing, that never happened in CI (in combination with the other small adaptation of this PR, because just this adaptation doesn't resolve it completely ). 

Manual testing (copyright @jughosta)

Steps to reproduce the error:
* Install all sample data.
* On Discover page, create an ad hoc data view kib* with @timetamp as the time field. Edit data view and set timestamp as the time field. Edit again and remove the time field. Edit again and set timestamp as the time field.

Reply:
Still fails for me. I think it can't find the data view by its new ID, not the previous one.

<img width="1240" alt="Screenshot 2024-11-18 at 10 26 23" src="https://github.com/user-attachments/assets/1c1b4b88-7478-4995-98d3-67d2388cdb13">

Reply:
I think both cases apply, previous id can't be found, and new id can't be found 😿

Reply:
I observed that in https://github.com/elastic/kibana/blob/5fa46caa5c8d2b92d86cc850188a33382a80fb69/x-pack/plugins/lens/public/editor_frame_service/editor_frame/state_helpers.ts#L474 it had previous values instead of the new one.

Reply:
seems the list of data view is not always provided to this checking function, leading to this error 

![CleanShot 2024-11-18 at 19 44 15](https://github.com/user-attachments/assets/4d59f398-f8ce-4669-bf12-a70fee3d5622)

🤯

Reply:
@jughosta I don't dare to say: I think I've found a hack to solve it, until you confirm by testing (when having time).

Here's another take on it:
https://github.com/elastic/kibana/pull/200687

Reply:
That's a great finding! I will take a look, thanks!